### PR TITLE
n5: prevent failure on missing group metadata

### PR DESCRIPTION
**Pull request by joshmoore**
In the N5 spec, https://github.com/saalfeldlab/n5#file-system-specification
point 3, the `{"n5": "$version"}` metadata is only expected at the root level.
Datasets missing the metadata on inner groups fail to load:

```
In [43]: list(zarr.hierarchy.Group(store=zarr.n5.N5Store("ex.n5")).groups())
...
ValueError: group not found at path 'setup0'
```

This changes the group parsing logic to not fail if the metadata is
missing. Suggestions welcome on how to detect the top group/array in
order to special case the parsing.

see https://github.com/tischi/i2k-2020-s3-zarr-workshop/issues/1#issuecomment-726224950  for the original report

TODO:
* [x] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] AppVeyor and Travis CI passes
* [ ] Test coverage is 100% (Coveralls passes)

### CODE REVIEW

File: zarr/n5.py [hunk 1]
```diff
@@ -304,13 +304,18 @@ def invert_chunk_coords(key):
...
 
 
 def group_metadata_to_zarr(group_metadata):
     '''Convert group metadata from N5 to zarr format.'''
-    del group_metadata['n5']
+    try:
+        del group_metadata['n5']
+    except KeyError:
+        # This only exists at the top level
+        pass
```
Comment (jakirkham) [inline zarr/n5.py:318]: Would suggest using `pop`. This just removes the key and returns the corresponding value. As it is a Python `dict`, this operation is pretty cheap.

```suggestion
    group_metadata.pop('n5')
```

Comment (joshmoore) [reply zarr/n5.py]: `.pop()` still throws a key error though, so I've re-added that handling.

Comment (joshmoore) [reply zarr/n5.py]:
```
>>> def a():
...     dict().pop(1)
...
>>> def b():
...     del dict()[1]
...
>>> import dis
>>> dis.dis(a)
  2           0 LOAD_GLOBAL              0 (dict)
              2 CALL_FUNCTION            0
              4 LOAD_METHOD              1 (pop)
              6 LOAD_CONST               1 (1)
              8 CALL_METHOD              1
             10 POP_TOP
             12 LOAD_CONST               0 (None)
             14 RETURN_VALUE
>>> dis.dis(b)
  2           0 LOAD_GLOBAL              0 (dict)
              2 CALL_FUNCTION            0
              4 LOAD_CONST               1 (1)
              6 DELETE_SUBSCR
              8 LOAD_CONST               0 (None)
             10 RETURN_VALUE

¯\_(ツ)_/¯
```

**Review by Carreau — APPROVED** [review]

File: zarr/n5.py (hunk 1 above)
Comment (Carreau) [reply zarr/n5.py]: `.pop('n5', None)` if you want a default value when the key does not exists.

Comment (jakirkham) [reply zarr/n5.py]: Yeah sorry forgot about `None`. Submitted PR ( https://github.com/zarr-developers/zarr-python/pull/655 ) which does this.

Comment (joshmoore) [reply zarr/n5.py]: For completeness:
```
>>> import dis
>>> def c():
...     dict().pop(1, None)
...
>>> dis.dis(c)
  2           0 LOAD_GLOBAL              0 (dict)
              2 CALL_FUNCTION            0
              4 LOAD_METHOD              1 (pop)
              6 LOAD_CONST               1 (1)
              8 LOAD_CONST               0 (None)
             10 CALL_METHOD              2
             12 POP_TOP
             14 LOAD_CONST               0 (None)
             16 RETURN_VALUE
```

**Pull request merged**